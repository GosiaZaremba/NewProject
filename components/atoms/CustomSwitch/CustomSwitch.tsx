import React, {
    forwardRef,
    useImperativeHandle,
    useRef,
    useState,
} from 'react';
import { Switch } from 'react-native';

export type Props = {
    trackColorFalse: string;
    trackColorTrue: string;
    thumbColorOn: string;
    thumbColorOff: string;
    testID?: string;
    ref: boolean;
};

export type CustomSwitchReference = {
    getValue: () => boolean;
    focus: () => void;
};

export const CustomSwitchWithReference: React.ForwardRefRenderFunction<
    CustomSwitchReference,
    Props
> = (
    { trackColorFalse, trackColorTrue, thumbColorOn, thumbColorOff, testID },
    ref
) => {
    const [remember, setRemember] = useState<boolean>(false);

    const switchReference = useRef<Switch>(null);

    const toggleSwitch = () => setRemember((remember) => !remember);

    useImperativeHandle(ref, () => ({
        getValue: () => {
            return remember;
        },
        focus: () => {
            switchReference?.current?.focus();
        },
    }));
    return (
        <Switch
            trackColor={{ false: trackColorFalse, true: trackColorTrue }}
            thumbColor={remember ? thumbColorOn : thumbColorOff}
            onValueChange={toggleSwitch}
            value={remember}
            ref={switchReference}
            testID={testID}
        />
    );
};

export const CustomSwitch = forwardRef(CustomSwitchWithReference);
