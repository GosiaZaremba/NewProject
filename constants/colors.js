export const Colors = {
  pink: '#fc296f',
  red: '#e63039',
  navy: '#39539b',
  blue: '#3872c3',
  grey: '#908D8E',
  light: '#F5F0F2',
};
