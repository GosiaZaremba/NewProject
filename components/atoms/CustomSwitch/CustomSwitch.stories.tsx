import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { CustomSwitch } from './CustomSwitch';
import { Colors } from '../../../constants/colors';

storiesOf('atoms/CustomSwitch', module)
    .addDecorator((story) => story())
    .add('default', () => (
        <CustomSwitch
            trackColorFalse={Colors.grey}
            trackColorTrue={Colors.grey}
            thumbColorOn={Colors.pink}
            thumbColorOff={Colors.grey}
        />
    ));
