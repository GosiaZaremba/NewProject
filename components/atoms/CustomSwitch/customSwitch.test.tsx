import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { CustomSwitch } from './CustomSwitch';
import { Colors } from '../../../constants/colors';

const mockOnPress = jest.fn();

describe('Switch Atom', () => {
    test('Component Snapshot', () => {
        const tree = renderer
            .create(
                <CustomSwitch
                    trackColorFalse={Colors.grey}
                    trackColorTrue={Colors.pink}
                    thumbColorOn={Colors.grey}
                    thumbColorOff={Colors.grey}
                    testID={'atom-switch'}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('ensure variants works correctly', () => {
        const { getByTestId } = render(
            <CustomSwitch
                trackColorFalse={Colors.grey}
                trackColorTrue={Colors.pink}
                thumbColorOn={Colors.grey}
                thumbColorOff={Colors.grey}
                testID={'atom-switch'}
            />
        );
        const customSwitch = getByTestId('atom-switch');

        // expect(customInput.props.value).toBe('123');
        // expect(customInput.props.placeholder).toEqual('Custom Input');
        // expect(customInput.props.secureTextEntry).toEqual(false);
        // expect(customInput.props.onChangeText()).toHaveReturned();
        console.log(customSwitch);
    });
});
