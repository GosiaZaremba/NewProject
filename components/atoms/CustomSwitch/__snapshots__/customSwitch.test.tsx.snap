// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Atom Component Snapshot 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#fc296f"
  style={
    Object {
      "height": 31,
      "width": 51,
    }
  }
  testID="atom-switch"
  thumbTintColor="#908D8E"
  tintColor="#908D8E"
  value={false}
/>
`;
